wordpress-mobile/WordPress-Android
# Implement domains list search logic

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/19173

### Description

This PR adds search (filtering) logic to the my domains screen.

To test:

Prerequisite: Please test this with an account which has several domains.

1. Navigate to the "My Domains" screen (Me -> Domains)
2. Wait for your domains to load
3. Type a search query into the input

Expect that the domains shown in the list are filtered based on the query. Only matching domains should be shown. A domain matches if the query string is present within either the `domain` or the `siteSlug` for the domain.

## Regression Notes
1. Potential unintended areas of impact
N/A

5. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

6. What automated tests I added (or what prevented me from doing so)
Unit tests were added for this logic.

PR submission checklist:


UI Changes testing checklist:
